examples/vae/main.py

```diff
@@ -0,0 +1,147 @@
+from absl import app
+from absl import flags
+
+import jax.numpy as jnp
+import numpy as np
+
+import jax
+from jax import nn as jnn
+from jax import random
+
+from flax import nn
+from flax import optim
+
+import tensorflow_datasets as tfds
+
+from utils import save_image
+
+
+EPS = 1e-8
+KEY = random.PRNGKey(0)
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_float(
+    'learning_rate', default=1e-3,
+    help=('The leanring rate for the Adam optimizer')
+)
+
+flags.DEFINE_integer(
+    'batch_size', default=128,
+    help=('Batch size for training')
+)
+
+flags.DEFINE_integer(
+    'num_epochs', default=10,
+    help=('Number of training epochs')
+)
+
+
+class Encoder(nn.Module):
+    def apply(self, x):
+        x = nn.Dense(x, 400, name='enc_fc1')
```

Comment:
the flax codebase has 2 spaces indentation instead of 4

Reply:
I changed the code to have 2 spaces.